src/org/sosy_lab/java_smt/solvers/bitwuzla/BitwuzlaNativeApiTest.java

```diff
@@ -0,0 +1,622 @@
+    assertThat(foo.kind()).isEqualTo(Kind.CONSTANT);
+
+    Term add = termManager.mk_term(Kind.FP_ADD, rm, two, one);
+    Term eq = termManager.mk_term(Kind.EQUAL, add, a);
+    Term neg = termManager.mk_term(Kind.NOT, eq);
+
+    // The type of add is fp (a bv add would be bv)
+    assertThat(add.sort().is_fp()).isTrue();
+    // eq is bool
+    assertThat(eq.sort().is_bool()).isTrue();
+    // neg is also bool
+    assertThat(neg.sort().is_bool()).isTrue();
+    assertThat(neg.sort().is_fun()).isFalse();
+
+    // Non-UF functions consist of a KIND and arguments.
+    // You can get the KIND w bitwuzla_term_get_kind() and the arguments in the correct order w
+    // bitwuzla_term_get_children()
+    assertThat(appliedFoo.sort().is_fun()).isFalse();
+    assertThat(add.sort().is_fun()).isFalse();
+    assertThat(eq.sort().is_fun()).isFalse();
+    assertThat(neg.sort().is_fun()).isFalse();
+
+    assertThat(rm.sort().is_rm()).isTrue();
+
+    Sort aSort = a.sort();
+    assertThat(aSort.is_bv()).isFalse();
+    assertThat(aSort.is_fp()).isTrue();
+    assertThat(aSort.is_fun()).isFalse();
+    assertThat(aSort.is_rm()).isFalse();
+    assertThat(aSort.is_bool()).isFalse();
+    assertThat(aSort.is_array()).isFalse();
+
+    assertThat(a.is_const()).isTrue();
+    assertThat(a.is_variable()).isFalse();
+    assertThat(a.is_value()).isFalse();
+    assertThat(a.kind()).isEqualTo(Kind.CONSTANT);
+
+    assertThat(a.toString()).isEqualTo("a");
+
+    Sort oneSort = one.sort();
+    assertThat(one.kind()).isEqualTo(Kind.VALUE);
+    assertThat(one.kind()).isNotEqualTo(Kind.VARIABLE);
+    assertThat(one.kind()).isNotEqualTo(Kind.CONSTANT);
+    assertThat(oneSort.is_bv()).isFalse();
+    assertThat(oneSort.is_fp()).isTrue();
+    assertThat(one.is_const()).isFalse();
+    assertThat(oneSort.is_fp()).isTrue();
+    assertThat(one.is_variable()).isFalse();
+    assertThat(oneSort.is_fun()).isFalse();
+    assertThat(oneSort.is_rm()).isFalse();
+    assertThat(oneSort.is_bool()).isFalse();
+    assertThat(oneSort.is_array()).isFalse();
+
+    assertThat(one.toString()).isEqualTo("(fp #b0 #b01111 #b0000000000)");
+  }
+
+  /*
+   * This serves as a testbed for indexed terms
+   */
+  @Test
+  public void testExtend() {
+    Sort bvSort8 = termManager.mk_bv_sort(8);
+    Sort bvSort10 = termManager.mk_bv_sort(10);
+    Term x = termManager.mk_const(bvSort8, "x");
+    Term y = termManager.mk_const(bvSort10, "y");
+    Term xExt = termManager.mk_term(Kind.BV_SIGN_EXTEND, x, 2);
+    Term xExtEqY = termManager.mk_term(Kind.EQUAL, xExt, y);
+    bitwuzla.assert_formula(xExtEqY);
+    Result res = bitwuzla.check_sat();
+    assertThat(res).isEqualTo(Result.SAT);
+
+    assertThat(xExt.num_children()).isEqualTo(1);
+    assertThat(xExt.num_indices()).isGreaterThan(0);
+    Vector_Int indices = xExt.indices();
+    assertThat(indices.size()).isEqualTo(1);
+    assertThat(indices.get(0)).isEqualTo(2);
+  }
+
+  // Todo:
```

Comment:
Do you remember what this test was meant to demonstrate? If I run it the result is SAT, which seems fine to me. Then again it uses `forall` instead of `exists`?

Reply:
I think i was just lazy and copied the `forall` test. I fixed it now and added model checking. Thanks for the reminder.